addons/addon-base-raas/packages/base-raas-services/lib/project/project-service.js

```diff
@@ -209,13 +224,74 @@ class ProjectService extends Service {
 
     if (restrict) return [];
 
-    // Future task: only return projects that the user has been associated with unless the user is an admin
-
     // Remember doing a scan is not a good idea if you billions of rows
-    return this._scanner()
+    let projects = await this._scanner()
       .limit(1000)
       .projection(fields)
       .scan();
+
+    if (this.isAppStreamEnabled) {
+      projects = await this.verifyAppStreamConfig(projects);
+    }
+
+    // Only return projects that the user has been associated with unless user is an admin
+    if (isAdmin(requestContext) || isSystem(requestContext)) return projects;
+
+    const by = _.get(requestContext, 'principalIdentifier.uid');
+    const retVal = await Promise.all(
+      _.map(projects, async proj => {
+        if (await this.verifyUserProjectAssociation(by, proj.id)) return proj;
+        return undefined;
+      }),
+    );
+    return _.filter(retVal, val => !_.isUndefined(val));
+  }
+
+  /**
+   * Returns the same object type as the input with an added isAppStreamConfigured boolean flag for each project entity
+   *
+   * IMPORTANT: The system context is used instead of the user's request context.
+   * This is because non-admin users do not have access to resources like indexes and awsAccounts,
+   * which are required for config verification. No other information from these resources are made available to non-admins
+   *
+   * @param input The Project entity, or list of Project entities
+   */
+  async verifyAppStreamConfig(input) {
+    try {
+      const systemContext = getSystemRequestContext();
+      const [awsAccountsService, indexesService] = await this.service(['awsAccountsService', 'indexesService']);
+      const accounts = await awsAccountsService.list(systemContext, {
+        fields: ['id', 'appStreamFleetName', 'appStreamSecurityGroupId', 'appStreamStackName'],
+      });
+      const indexes = await indexesService.list(systemContext, { fieldsToGet: ['id', 'awsAccountId'] });
+      const accountIdsWithAppStream = _.map(
+        _.filter(
+          accounts,
+          account =>
+            !_.isUndefined(account.appStreamFleetName) &&
```

Comment:
Thoughts on making the operation an `||` instead of `&&`? I believe if any of those fields are missing, AppStream will not function correctly when trying to generate an AppStream link.

Reply:
You're right. Making the change

Reply:
We can't, my unit test caught it :)

If any of those fields are missing we want to exclude that account from the array. Therefore we need to use `&&` instead of `||`, because here we're only accepting accounts with _None_ of these 3 fields set to `undefined`.